### ISSUE TITLE
[RFC] Not take SIGINT handler in embedded mode

### ISSUE THREAD

**Issue by discord9** [RustPython/RustPython#4408]
## Summary

Currently, RustPython takes SIGINT handler regradless of whether it's in repl mode or embedded mode, however in embedded mode, it's reasonable that the embedding application might not want RustPython to take the SIGINT handler, maybe we could add a option to disable SIGINT handler in embedded mode?

## Detailed Explanation
 
Upon inited, RustPython takes SIGINT handler, the related code is basically as follow:
```
vm::stdlib::make_module->
signal::make_module ->
init_signal_handlers->
libc::signal(SIGINT,<custom handler>)
```

## Drawbacks, Rationale, and Alternatives

Could interfer with code that depend on SIGINT/KeyBoardInterrput, so it might need to be a opt-out option

## Unresolved Questions

-

**Comment by discord9** [comment]
any ideas? @coolreader18

**Comment by discord9** [comment]
Closed by #4475
